### PR TITLE
Add option to render using the dashboard timezone

### LINKED ISSUE #135: [FeatureRequest] New time zone option to follow dashboard's time zone selection
To allow all dates and times on a dashboard to sync up, it would be nice for the clock to be able to follow the time zone selected in the dashboard's datetime picker.

Screenshot 2023-04-17 150754

### PR DESCRIPTION
**Pull request by academo**
Adds a new option in the panel to use the same timezone as the dashboard 
image

The dashboard timezone is the same as the browser by default but can be changed in the time range selection

image

1. Change the time range
2. Change time settings
3. Select a different time zone

With this option selected, not only the time will display the dashboard timezone but it'll be also take in consideration for count ups and count downs.

Closes https://github.com/grafana/clock-panel/issues/135

### CODE REVIEW

File: src/options.ts [hunk 1]
```diff
@@ -199,10 +199,13 @@ function getVariableOptions() {
...
-    return { label: n, value: n };
-  });
-  timezones.unshift({ label: 'Default', value: '' });
+  const timezones = [
+    { label: 'Default (Browser)', value: '' },
+    { label: 'Same as Dashboard', value: 'dashboard' },
+    ...getTimeZoneNames().map((n) => {
+      return { label: n, value: n };
+    }),
+  ];
```
Comment (mckn) [inline src/options.ts:208]: What do you think about changing the UI for this a bit? For me it would make sense to have two options:

1. Same as dashboard
2. Custom for panel

When a user selects the first one we use the same as the dashboard is configured to use. When the user selects the other one we ask them for the time zone that they want to use.

When it comes to the options of that selector I think we should try to use the same list as we are using in the TimePicker.

So basically:

- Default
- Browser Time
- UTC
- ...time zones

Default can be configured on two levels. Either on the organisational level or for the user account.

Browser Time is what ever timezone your browser is set to.

**Review by mckn — COMMENTED** [review]
I think the code looks good! But not sure about the time zone options and the UI of that editor. Please see the other comment for more details.

File: src/options.ts (hunk 1 above)
Comment (academo) [reply src/options.ts:208]: @mckn That's a good option but It'll probably require some migration from one version of the panel to the other since we currently store the timezone as a single field.

I also like the idea of keeping a familiar UI for people upgrading the panel. 

Regarding the "Default" in the second list. where can I get it from? In the props we only get two timezones: The one that comes from the dashboard and the one that the user could have selected from the selector.

Comment (mckn) [reply src/options.ts:208]: > @mckn That's a good option but It'll probably require some migration from one version of the panel to the other since we currently store the timezone as a single field.
> 
Yeah, I gave this a bit of a more thought and you are correct. Probably not worth changing too much in the UI. Stick with what you have!

> I also like the idea of keeping a familiar UI for people upgrading the panel.
> 
> Regarding the "Default" in the second list. where can I get it from? In the props we only get two timezones: The one that comes from the dashboard and the one that the user could have selected from the selector.

Hmm, if you call this function, without any options, `getTimeZone()`  from `grafana/data` you will get the currently selected default time zone.

Comment (mckn) [reply src/options.ts:208]: https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/datetime/common.ts#L59C14-L59C25

Comment (academo) [reply src/options.ts:208]: @mckn if you are ok with not changing the UI. Should we merge this PR as is? Or should we add an option to "use default timezone"?

The whole "default" thing gets now more complicated. We have now the Browser Time. The Dashboard Time. The User-defined time (in the panel) and the grafana default time.

Comment (mckn) [reply src/options.ts:208]: I think we can keep the UI but I think we should do one of the following:

- Remove the `Default (Browser)` since the default can be an other timezone than `browser` so that label is not correct.
- Rename `Default (Browser)` to `Default` and add another value called `Browser Time`. This correlates with the options in the Time Picker.

Comment (academo) [reply src/options.ts:208]: @mckn In this specific case the "default" in this plugin is indeed the browser. so `Default (browser)` is the correct one.

There's currently no logic getting the `Default` as the one coming from the grafana configuration. The plugin had always used the word "Default" to refer to the browser time, that's why I added the clarification in this PR.

Comment (mckn) [reply src/options.ts:208]: Ah, ok ok - then I would suggest to change the name from `Default` to `Browser Time` to align it with the time picker. It would make it much clearer and consistent.

Comment (mckn) [reply src/options.ts:208]: This is a perfect scenario where @sympatheticmoose usually have great input so I'm adding him here.

Comment (sympatheticmoose) [reply src/options.ts:208]: Sorry this has taken so long to get to. 

Based on the above I would be tempted to rename to `Browser Time` so its consistent with the timezone picker in the dashboard. Hadn't realised that Default was actually "a thing" and one that we don't support currently (nor am I convinced that adding support for it is urgent). 

The middle ground would be `Browser Time (Default)` but if we then want to extend with support for `Default` at a later stage, this becomes more confusing.

Comment (academo) [reply src/options.ts]: @sympatheticmoose @mckn I changed it to `Browser Time` that makes mores sense. No migrations required for this case since is just a label change.

**Review by sympatheticmoose — APPROVED** [review]
🚢

File: src/options.ts (hunk 1 above)
Comment (mckn) [reply src/options.ts]: Amazing! Great work on this

**Review by mckn — APPROVED** [review]
LGTM!
